dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java

```diff
@@ -93,11 +92,14 @@ public void join(WorkerInfo workerInfo) throws IOException {
     byte[] serializedEntity = entity.serialize();
     // If there's existing entry, check if it's me.
     if (existingEntityBytes != null) {
-      // It's not me, something is wrong.
+      // It's not me, or not the same me.
       if (!Arrays.equals(existingEntityBytes, serializedEntity)) {
-        // Might be regression of different formatted value of workerinfo is registered.
-        throw new AlreadyExistsException(
-            "Some other member with same id registered on the ring, bail.");
+        // In k8s this might be bcos worker pod restarting with the same worker identity
+        // but certain fields such as hostname has been changed. Register to ring path anyway.
+        LOG.warn("Same worker entity found bearing same workerid,"
```

Comment:
If the serialized forms don't compare equal, it doesn't mean the two have different worker IDs. Could be different network address, or anything that's in `WorkerServiceEntity`.

Reply:
yeah exactly, we found the key already exists, so now we have to overwrite it with a different WorkerServiceEntity content, before it doesnt allow doing so.

Reply:
Ok I see what you are trying to achieve. If the serialized forms are equal, then you don't have to overwrite anything. Makes sense.